This patch adds the current XXH3 checksum algorithms to rsync. They
are not yet finalized, so do NOT use this in a distributed rsync yet!

To use this patch, run these commands for a successful build:

    patch -p1 <patches/xxh3.diff
    ./configure                         (optional if already run)
    make

based-on: 00dd50a00c00da4824b3aa5fe2b2f82baffca071
diff --git a/checksum.c b/checksum.c
--- a/checksum.c
+++ b/checksum.c
@@ -26,11 +26,13 @@
  * with this program; if not, visit the http://fsf.org website.
  */
 
+#define XXH_INLINE_ALL /* define to test pre-release XXH3 & XXH128 -- CURRENTLY UNSTABLE */
+
 #include "rsync.h"
 
 #ifdef SUPPORT_XXHASH
 #include "xxhash.h"
-# if XXH_VERSION_NUMBER >= 800
+# if XXH_VERSION_NUMBER >= 800 || defined XXH_INLINE_ALL
 #  define SUPPORT_XXH3 1
 # endif
 #endif
